api/tests/unit/features/test_unit_features_views.py

```diff
@@ -1264,3 +1270,153 @@ def test_cannot_update_feature_of_a_feature_state(
     assert (
         response.json()["feature"][0] == "Cannot change the feature of a feature state"
     )
+
+
+def test_create_segment_override__using_simple_feature_state_viewset__allows_manage_segment_overrides(
+    staff_client: APIClient,
+    with_environment_permissions: Callable[[list[str], int], None],
```

Comment:
```suggestion
    with_environment_permissions: Callable[[list[str]], None],
```

Not precisely accurate but correct

Reply:
Well, technically, I guess it should be `Callable[[list[str], int | None], None]` based on the definition: 

```python
@pytest.fixture()
def with_environment_permissions(
    environment: Environment, staff_user: FFAdminUser
) -> typing.Callable[[list[str], int], None]:
    """
    Add environment permissions to the staff_user fixture.
    Defaults to associating to the environment fixture.
    """

    def _with_environment_permissions(
        permission_keys: list[str], environment_id: typing.Optional[int] = None
    ) -> UserEnvironmentPermission:
        environment_id = environment_id or environment.id
        uep, __ = UserEnvironmentPermission.objects.get_or_create(
            environment_id=environment_id, user=staff_user
        )
        uep.permissions.add(*permission_keys)

        return uep

    return _with_environment_permissions
```

Reply:
Fixed [here](https://github.com/Flagsmith/flagsmith/pull/2990/commits/64261ae9939e2e37fcd1d0f75273816d5cf76f94)